iox-#743 Review configuration guide (#1282)
eclipse-iceoryx/iceoryx: doc/website/advanced/configuration-guide.md

## :material-cog: CMake switches for configuring iceoryx_posh build

There are several configuration options set by default when iceoryx_posh is build.
These options adjust the global maximum amount of resources like Publisher and
Subscriber Ports which can have a huge impact on the memory footprint of iceoryx

> since the define the size of the management structures

Reviewer: ```suggestion
since they define the size of the management structures
```

 | `IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY` | Maximum number of chunks a subscriber can take in parallel|
 | `IOX_MAX_INTERFACE_NUMBER` | Maximum number of interface ports which are used by gateways |

Have a look at 
[IceoryxPoshDeployment.cmake](https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_posh/cmake/IceoryxPoshDeployment.cmake)
for the default values of the constants.

!!! hint
    With the default values set, the size of `iceoryx_mgmt` is ~64.5 MByte. You
    can reduce the size by decreasing the values from the table via the CMake 
    options. The current values are printed in the CMake stage when building iceoryx.

Example:

```bash
cmake -Bbuild -Hiceoryx_meta -DIOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY=64
```

With that change, the footprint of the management segment is reduced to ~52.7 MBytes.
For larger use cases you can increase the value to avoid that samples are dropped
on the subscriber side (see also [#615](https://github.com/eclipse-iceoryx/iceoryx/issues/615)).

## :material-memory: Configuring Mempools for RouDi

RouDi supports several shared memory segments with different access rights, to

> limit the read and write access between different applications. Memory pools are
> then managing the those segments and organizing the user payload data required

Reviewer: ```suggestion
limit the read and write access between different applications. Memory pools manage those segments and organize the user payload data required
```

for communication.

!!! note
    Actually only the chunk-payload size is configured and the size of the
    `ChunkHeader` will be added to the configured size. If a user-header or a
    user-payload alignment larger than 8 is used, the available size for the
    user-payload will be smaller than the configured chunk-payload since some
    space is needed for the other functionality.
    Please have a look at the 

>     [chunk_header.md](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/design/chunk_header.md)

Reviewer: ```suggestion
    [chunk_header.md](https://github.com/eclipse-iceoryx/iceoryx/blob/v2.0.0/doc/design/chunk_header.md)
```

Author: Since it is merged to master I would leave it has it is.

### :material-file-cog: Dynamic configuration

One way is to read a configuration dynamically during the startup of RouDi.
Using the TOML Config in RouDi is not mandatory for configuring segments and
mempools, but a comfortable alternative.

To enable the TOML config in iceoryx, the CMake option `-DTOML_CONFIG=ON` must

> be enabled (this is the default setting).

Reviewer 2: This reads somehow weird. If it is the default setting why writing about enabling it. What do you think of
```
The CMake option `-DTOML_CONFIG`, which is `ON` by default, enables the TOML config for iceoryx.
```

count = 100
```

When no configuration file is specified a hard-coded version similar to the 

> [default config](https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_posh/etc/iceoryx/roudi_config_example.toml)

Reviewer: ```suggestion
[default config](https://github.com/eclipse-iceoryx/iceoryx/blob/v2.0.0/iceoryx_posh/etc/iceoryx/roudi_config_example.toml)
```

Author: Since it is merged to master I would leave it has it is.

will be used.

### Static configuration

Another way is to have a static configuration that is compiled into the roudi application.
As a consequence you have to recompile your RouDi application if you want to change
your config (not the `iceoryx_posh_roudi` lib).
You can create your custom configuration in the `main()` function where you then pass
it as constructor argument to the RouDi instance.
In the cmake file entry of the custom RouDi executable you need to ensure that it
is **not** linking against `iceoryx_posh_config` to ensure using the static configuration.

An example of a static config can be found 

> [here](https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_examples/iceperf/roudi_main_static_config.cpp).

Reviewer: ```suggestion
[here](https://github.com/eclipse-iceoryx/iceoryx/blob/v2.0.0/iceoryx_examples/iceperf/roudi_main_static_config.cpp).
```

Author: Since it is merged to master I would leave it has it is.
